cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java

```diff
@@ -357,6 +368,80 @@ public void testTableWithPermissions() throws IOException {
     Assert.assertFalse(admin.exists());
   }
 
+  @Test
+  public void testScannerCache() throws Exception {
+    String tableName = "scanCache";
+    // note: it appears that HBase only enforces the scanner timeout after 10 seconds.
+    // setting it to 3 seconds does not mean it will actually fsail after 3 sweconds.
+    // therefore we have to cross the 10 seconds. here: 1200 times 10ms sleep.
+    int numRows = 1200;
+    DatasetAdmin admin = getTableAdmin(CONTEXT1, tableName);
+    admin.create();
+    try {
+      // write some rows and commit
+      Transaction tx1 = txClient.startShort();
+      Table myTable1 = getTable(CONTEXT1, tableName);
+      ((TransactionAware) myTable1).startTx(tx1);
+      for (int i = 0; i < numRows; i++) {
+        myTable1.put(new Put("" + i, "x", "y"));
+      }
+      Assert.assertTrue(txClient.canCommit(tx1, ((TransactionAware) myTable1).getTxChanges()));
+      Assert.assertTrue(((TransactionAware) myTable1).commitTx());
+      Assert.assertTrue(txClient.commit(tx1));
+
+      try {
+        testScannerCache(numRows, tableName, null, null, null);
+        Assert.fail("this should have failed with ScannerTimeoutException");
+      } catch (Exception e) {
+        // we expect a RuntimeException wrapping an HBase ScannerTimeoutException
```

Comment:
I don't know why but that's what we get. Throwables.propagate() all over the place...